R/qenv-constructor.R

```diff
@@ -10,9 +10,31 @@
 #' `qenv()` instantiates a `qenv` with an empty environment.
 #' Any changes must be made by evaluating code in it with `eval_code` or `within`, thereby ensuring reproducibility.
 #'
+#'
 #' `new_qenv()` (`r badge("deprecated")` and not recommended)
 #' can instantiate a `qenv` object with data in the environment and code registered.
 #'
+#' @section Environment:
+#'
+#' The `qenv` object behaves as an environment that is locked and can be used as
+#' as a argument to with many functions that accept `environment`, among the
+#' most relevant are: `names()`, `ls()`, `get()`, `exists()`, `parent.env()`,
+#' `{l,s,v}apply` family, `local`, `as.environment()`, `is.environment()`, `as.list()`, ...
+#'
+#' `qenv` should not be used with `cbind()` and `env.profile()` functions as it
+#' has unexpected behavior.
+#' Instead, `get_env()` or `as.environment()` should be used before calling any
+#' problematic function.
+#'
+#' Similarly, `rlang` functions related to environments cannot be used directly
+#' with `qenv` and should be used with `teal.code::get_env()`/`as.environment()`.
+#'
+#' ```r
+#' q <- qenv()
+#' rlang::env_clone(as.environment(q))
+#' rlang::env_clone(get_env(q))
+#' ```
```

Comment:
I think "_We don't recommend using any function outside of the `teal.code` exports and these mentioned above_" is enough. Explaining that something doesn't work is adding more content to the simple statement. Also I don't see the point of guiding users what to do when `rlang` fails.

```suggestion
```